### PR TITLE
Attempt to fix how we import cloned contracts and set verify args

### PR DESCRIPTION
**Pull request by jflatow**
Also add some more options to control deploy task:
 `--no-deploy`: skip actually deploying anything
 `--no-verify-impl`: don't verify the Comet impl

### CODE REVIEW

File: tasks/deployment_manager/task.ts [hunk 1]
```diff
@@ -48,10 +48,12 @@ async function runMigration<T>(
 
 task('deploy', 'Deploys market')
   .addFlag('simulate', 'only simulates the blockchain effects')
+  .addFlag('noDeploy', 'skip the actual deploy step')
```
Comment (kevincheng96) [inline tasks/deployment_manager/task.ts:51]: IMO, it's more intuitive if we create a new task in this file called `verify` that only runs the verify process (the `deploy` task will still call verify as well). Running `deploy` with a `noDeploy` flag feels kind of unintuitive.

**Review by kevincheng96 — COMMENTED** [review]
Verify fixes look promising!

File: tasks/deployment_manager/task.ts (hunk 1 above)
Comment (kevincheng96) [reply tasks/deployment_manager/task.ts:51]: Yeah sure thing, a separate PR is fine

File: plugins/import/import.ts [hunk 2]
```diff
@@ -78,21 +88,68 @@ async function getContractCreationCode(network: string, address: string) {
   return matches[0][1];
 }
 
+function parseSources({ source, contract, optimized, optimizationRuns }: EtherscanData) {
```
Comment (hayesgm) [inline plugins/import/import.ts:91]: This is crazy, but hazaah!

**Review by hayesgm — APPROVED** [review]

File: plugins/import/import.ts [hunk 3]
```diff
@@ -78,21 +88,68 @@ async function getContractCreationCode(network: string, address: string) {
...
+    return {
+      language: 'Solidity',
+      settings: {
+        optimizer: {
+          enabled: optimized,
+          runs: optimizationRuns,
+        }
+      },
+      sources: {
+        [`contracts/${contract}.sol`]: {
```
Comment (scott-silver) [inline plugins/import/import.ts:119]: should this be:

````
`contracts/${contract}.sol:${contract}`
```

?

**Review by scott-silver — APPROVED** [review]

File: plugins/deployment_manager/ManualVerify.ts [hunk 4]
```diff
@@ -54,14 +55,15 @@ export async function manualVerifyContract(
     settings.optimizer.runs = 1000000;
   }
 
-  let request = toVerifyRequest({
+  const args = await encodeArguments(contractMetadata.abi, sourceName, contractFQN, deployArgs);
```
Comment (kevincheng96) [inline plugins/deployment_manager/ManualVerify.ts:58]: Hmm, I'm a bit confused about this part. Shouldn't we just be encoding the `deployArgs`?

Comment (kevincheng96) [reply plugins/deployment_manager/ManualVerify.ts:58]: Actually nevermind, just saw that this is a `hardhat-etherscan` function that takes these specific args

**Review by kevincheng96 — APPROVED** [review]
👍

File: plugins/deployment_manager/ManualVerify.ts (hunk 4 above)
Comment (jflatow) [reply plugins/deployment_manager/ManualVerify.ts:58]: Isn't that what `encodeArguments` does?

Comment (jflatow) [reply plugins/deployment_manager/ManualVerify.ts:58]: https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-etherscan/src/ABIEncoder.ts#L10

File: plugins/import/import.ts (hunk 3 above)
Comment (jflatow) [reply plugins/import/import.ts:119]: Not according to etherscan, I think

File: plugins/deployment_manager/ManualVerify.ts (hunk 4 above)
Comment (jflatow) [reply plugins/deployment_manager/ManualVerify.ts:58]: > Actually nevermind, just saw that this is a `hardhat-etherscan` function that takes these specific args

Ah this comment wasn't showing before 👍

**Comment by jflatow** [comment]
Merged by #565

**Pull request closed without merging**